src/chunk.c

```diff
@@ -1620,6 +1620,7 @@ chunk_tuple_found(TupleInfo *ti, void *arg)
 	chunk->hypertable_relid = ts_hypertable_id_to_relid(chunk->fd.hypertable_id, false);
 
 	chunk->relkind = get_rel_relkind(chunk->table_id);
+	chunk->amoid = ts_get_rel_am(chunk->table_id);
```

Comment:
Some observations on this:

1. I think you should add this to `ts_chunk_scan_by_chunk_ids` as well. Probably a good idea to do a catalog lookup once, because the relkind and amoid are in the same pg_class catalog. Maybe you can make a function and use it here and there as well.

2. It's not failing, so this means it's dead code now. Let's add some assertion that this is not null to improve test coverage.

Reply:
Added to `ts_chunk_scan_by_chunk_ids` :+1: 

Note sure what kind of assertions you are asking for.

The table AM used can be InvalidOid on, e.g., OSM chunks. So, it is not clear what I should assert.